src/EditorFeatures/Core.Wpf/Options/LegacyEditorConfigDocumentOptionsProvider.cs

```diff
@@ -135,6 +149,21 @@ private void ReleaseContext_NoLock(Task<ICodingConventionContext> contextTask)
 
         public async Task<IDocumentOptions> GetOptionsForDocumentAsync(Document document, CancellationToken cancellationToken)
         {
+            if (_enabled is null)
+            {
+                var experimentationService = await _experimentationServiceFactory.GetExperimentationServiceAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Why isn't this still a deadlock risk if somebody were to ask for options on the UI thread and we hadn't already initialized it? Or are we taking a tactic that it's still not correct but "good enough"?

Reply:
> Why isn't this still a deadlock risk if somebody were to ask for options on the UI thread and we hadn't already initialized it?

The deadlock occurred because MEF isn't JTF aware. We moved the part with thread dependencies outside of the MEF construction path where code is expected to be JTF aware.

Reply:
📝 There is still a deadlock risk if users block on the completion of this asynchronous method. This is the same deadlock risk that applies to all cases in Roslyn where code blocks on the completion of an asynchronous operation (no cases are proven correct, therefore the only safe assumption is all cases are potential deadlock scenarios).

Reply:
Sure, but right now the only callers of this method are Roslyn code, so this would have to follow Roslyn threading rules instead of JTF rules. "Code is expected to be JTF aware" might be a guideline, but it's patently not the case here.